File: ScoreSystemServer/ScoreSystemServer/src/main/java/com/score/system/entity/user/StudentDTO.java
package com.score.system.entity.user;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
public class StudentDTO {
    @NotBlank(message = "学生名字不能为空")
    private String name;
    private String userName;
    private String password;
    @NotBlank(message = "学号不能为空")
    private String studentNumber;
    private Integer classId;
    private Long state;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate enrollmentDate;
    @NotNull(message = "科目组合ID不能为空")
    @Min(value = 1, message = "选科ID错误")
    private Long subjectGroupId;
    @NotNull(message = "选修课1不能为空")
    @Min(value = 1, message = "选修课1 ID错误")
    private Long electiveCourse1Id;
    @NotNull(message = "选修课2不能为空")
    @Min(value = 1, message = "选修课2 ID错误")
    private Long electiveCourse2Id;
}

File: ScoreSystem/ScoreSystem/ScoreSystem/Model/StudentDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace ScoreSystem.Model
{
    public class StudentDTO
    {
        private string name;
        private string userName;
        private string password;
        private string studentNumber;
        private int classId;
        private int state;
        private DateTime enrollmentDate;
        private int subjectGroupId;
        private int electiveCourse1Id;
        private int electiveCourse2Id;
        private int year;

        public string Name { get => name; set => name = value; }
        public string UserName { get => userName; set => userName = value; }
        public string Password { get => password; set => password = value; }
        public string StudentNumber { get => studentNumber; set => studentNumber = value; }
        public int ClassId { get => classId; set => classId = value; }
        public int State { get => state; set => state = value; }
        public DateTime EnrollmentDate { get => enrollmentDate; set => enrollmentDate = value; }
        public int SubjectGroupId { get => subjectGroupId; set => subjectGroupId = value; }
        public int ElectiveCourse1Id { get => electiveCourse1Id; set => electiveCourse1Id = value; }
        public int ElectiveCourse2Id { get => electiveCourse2Id; set => electiveCourse2Id = value; }
        public int Year { get => year; set => year = value; }
    }
}
